### PR TITLE
chore: bump local pgsql from 11 to 14

### PR DESCRIPTION
**Pull request by lecoqjacob**
## Description of change
Bumps the local version of pgsql to match our deployed versions 

Addresses: #1072

### CODE REVIEW

**Review by iulianbarbu — DISMISSED** [review]
Thanks! lgtm

**Comment by iulianbarbu** [comment]
@lecoqjacob it seems like 1 test is failing. Can you reproduce it locally?

**Comment by lecoqjacob** [comment]
Password hashing changed in v14 of psql to use sham-sha-256.

https://www.percona.com/blog/postgresql-14-and-recent-scram-authentication-changes-should-i-migrate-to-scram/

File: cargo-shuttle/src/provisioner_server.rs [hunk 1]
```diff
@@ -333,7 +333,7 @@ fn db_type_to_config(db_type: Type) -> EngineConfig {
     match db_type {
         Type::Shared(SharedEngine::Postgres) => EngineConfig {
             r#type: "shared_postgres".to_string(),
-            image: "docker.io/library/postgres:11".to_string(),
```
Comment (chesedo) [inline cargo-shuttle/src/provisioner_server.rs:336]: This tries to match the same version that is being provisioned for RDS instances to make local and prod identical. I might have missed when the RDS instances also got bumped?

**Review by chesedo — COMMENTED** [review]
We just have to verify the version stays in lockstep with the versions used for RDS

File: cargo-shuttle/src/provisioner_server.rs (hunk 1 above)
Comment (oddgrd) [reply cargo-shuttle/src/provisioner_server.rs:336]: I checked in the console now and a bit more than half of our RDS instances are at 14 (the newer ones), and the older half is at 13. I don't see any below 13, and the most recent instance is actually 15. :smile: We don't explicitly set the version in provisioner when we provision RDS dbs, so perhaps it just uses `latest` :thinking:

**Review by oddgrd — APPROVED** [review]
Thanks! Apologies for the delay.

**Pull request merged**